chore(docs): Add --api_host to readme and reveal in --help output

evanderkoogh opened the pull request:
## Which problem is this PR solving?

Make it clear how to use honeymarker in other regions.

## Short description of the changes

+ Added --api_host to the documentation

+ Also, reveal the api_host command line option now that there is more than one public API endpoint available (US & EU at the moment).
  + remove hidden:"true"
  + add a description for --help output
  + bonus: accept a value via environment variable like most of our other tools that accept an API endpoint option

evanderkoogh committed:
Add --api_host to readme

Make it clear how to use honeymarker in other regions.

robbkidd reviewed (approved):
👍🏻  The [api_host flag](https://github.com/honeycombio/honeymarker/blob/d5860e55cc0ca4ca59cbb10bc6e2e682c09111a6/main.go#L20) is definitely there. 

There's a `hidden` option on it that predates our multiple public instances (now 🇺🇸 & 🇪🇺) of the API endpoint. We should probably also remove it to make this flag appear in help output.

![I cannot tell you that. It's confidential.](https://github.com/user-attachments/assets/781605d4-8f98-4b46-95b5-987085cc56cb)

cewkrupa reviewed (approved):
Thank you!

robbkidd committed:
reveal the api_host CLI flag

We should stop hiding the api_host command line option now that there is
more than one public API endpoint available (US & EU at the moment).

+ remove hidden:"true"
+ add a description for --help output
+ bonus: accept a value via environment variable like most of our other
  tools that accept an API endpoint option

robbkidd committed:
no change - bang the set to see if Circle gets unstuck
